src/projectile.cpp

```diff
@@ -1176,7 +1176,22 @@ static void proj_ImpactFunc(PROJECTILE *psObj)
 		psObj->born = gameTime;
 
 		static GridList gridList;  // static to avoid allocations.
-		gridList = gridStartIterate(psObj->pos.x, psObj->pos.y, psStats->upgrade[psObj->player].radius);
+		bool useDestinationPos = false;
+		Vector3i targetPos;
+
+		if (psObj->psDest != nullptr && psObj->psDest->type == OBJ_DROID)
+		{
+			useDestinationPos = true;
+			targetPos = castDroid(psObj->psDest)->pos.xyz();
+			//If projectile impacts a droid start the splash damage from the center of the it.
+			gridList = gridStartIterate(targetPos.x, targetPos.y, psStats->upgrade[psObj->player].radius);
+		}
+		else
+		{
+			//Else use whatever location the projectile impacts at as the splash center.
+			gridList = gridStartIterate(psObj->pos.x, psObj->pos.y, psStats->upgrade[psObj->player].radius);
+		}
+
```

Comment:
Something like
```c++
// If projectile impacts a droid start the splash damage from the center of it, else use whatever location the projectile impacts at as the splash center.
auto destDroid = castDroid(psObj->psDest);
Vector3i targetPos = destDroid != nullptr ? destDroid->pos : psObj->pos;

static GridList gridList;  // static to avoid allocations.
gridList = gridStartIterate(targetPos.x, targetPos.y, psStats->upgrade[psObj->player].radius);
```
and later
```c++
if (useSphere && !Vector3i_InSphere(psCurr->pos, targetPos, psStats->upgrade[psObj->player].radius))
```
may be cleaner.

Reply:
Yep, much better.